From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Herman van Hazendonk <github.com@herrie.org>
Date: Sun, 10 Jul 2022 22:56:53 -0400
Subject: [PATCH] lib/build_OID_registry: don't embed the build path in output

Kbuild invokes this script as "perl $(srctree)/lib/build_OID_registry",
so $0 is the absolute path to the script whenever the kernel is built
out-of-tree (O=), which is what OE/Yocto always does. That absolute
path, which contains ${WORKDIR}/${TMPDIR}, ends up in the "Automatically
generated by" comment of lib/oid_registry_data.c and trips the
"buildpaths" QA check on the -src package.

Strip the $srctree prefix (exported by the top-level Makefile) from $0
before printing it, the same fix as upstream commit 0f586f4ee8ad
("lib/build_OID_registry: fix reproducibility issues"), which stripped
the path via File::Basename instead of a regex substitution.

[Herman: adapted for pre-abs_srctree kernel]

Originally by Bruce Ashfield <bruce.ashfield@gmail.com>, taken from shr-distribution/linux tissot/4.9/halium-9.0 (663b395111c4), where tissot builds with it.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
Upstream-Status: Backport
---
 lib/build_OID_registry | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/build_OID_registry b/lib/build_OID_registry
index 5d982721736..db911077847 100755
--- a/lib/build_OID_registry
+++ b/lib/build_OID_registry
@@ -39,7 +39,7 @@ close IN_FILE || die;
 #
 open C_FILE, ">$ARGV[1]" or die;
 print C_FILE "/*\n";
-print C_FILE " * Automatically generated by ", $0, ".  Do not edit\n";
+print C_FILE " * Automatically generated by ", $0 =~ s#^\Q$ENV{'srctree'}/\E##r, ".  Do not edit\n";
 print C_FILE " */\n";
 
 #
